framework/include/problems/SubProblem.h

```diff
@@ -288,39 +288,6 @@ class SubProblem : public Problem
    */
   virtual void clearActiveElementalMooseVariables(THREAD_ID tid);
 
-  /**
-   * Record and set the material properties required by the current computing thread.
-   * @param mat_prop_ids The set of material properties required by the current computing thread.
-   *
-   * @param tid The thread id
-   */
-  virtual void setActiveMaterialProperties(const std::set<unsigned int> & mat_prop_ids,
-                                           THREAD_ID tid);
-
```

Comment:
Never mind. Explained in the commit message.

Reply:
I'm copying this from one of my commit messages:
> All the material storage in in FEProblemBase. It is not very safe to
have the material property APIs down in SubProblem because a user of the
APIs could call on the displaced problem and then when it is time to
reinit the materials, it's possible we check and see that there are no
active material properties on the FEProblemBase and then we don't reinit the 
materials that are *only* ever held by `FEProblemBase`

Is that a clear enough explanation? A "user of the APIs" happened to be me when I was trying to convert all calls on `fe_problem` to `subproblem`

Reply:
Yeah, I think it's clear and that the change appears to make problem API usage more robust. I didn't see the description that you pasted initially.

Reply:
didn't reload the page, so I didn't see your second comment before I made mine! 😄